docs/tutorials/troubleshooting/deploy-vm.md

```diff
@@ -0,0 +1,54 @@
+### Describe Virtual Machine k8s resource
+After you ssh into the supervisor cluster control plane VM, run `kubectl describe vm <vm-name> -n <namespace-name>` command. Check `VM.Status.Conditions` as seen in sample output below.
+
+```console
+$ kubectl describe vm <vm-name> -n <namespace-name>
+
+...
+Status:
+  Bios UUID:              4239fba9-7390-1130-478a-e559a25f91e6
+  Change Block Tracking:  false
+  Conditions:
+    Last Transition Time:  2022-07-06T00:43:47Z
+    Status:                Unknown
+    Type:                  GuestCustomization
+    Last Transition Time:  2022-07-06T00:42:41Z
+    Status:                True
+    Type:                  VirtualMachinePrereqReady
+    Last Transition Time:  2022-07-06T00:43:47Z
+    Message:               VMware Tools is not running
+    Reason:                VirtualMachineToolsNotRunning
+    Severity:              Error
+    Status:                False
+    Type:                  VirtualMachineTools
+...
+```
+
+## Error and fix
+This section focuses on leveraging the information provided by the `VM.Status.Conditions` field to diagnose the root cause of errors and provide solutions. By analyzing which conditions showed `Status` as `False`/`Unknown` and `Reason`, you can troubleshoot issues depending on below scenarios.
+
+### VirtualMachinePrereqReady
+- Reason `VirtualMachineClassNotFound`/`VirtualMachineClassBindingNotFound`: this means that the VM class has not been associated to the Supervisor namespace.
+    - **Fix**: Attach VM Class to your namespace. Refer to [VMClass](https://vm-operator.readthedocs.io/en/stable/concepts/workloads/vm-class/) for details.
+- Reason `VirtualMachineImageNotFound`: this means that the VirtualMachine Image doesn't exist in the supervisor cluster, which means that this VM image doesn't exist in the vCenter, or the content library which it belongs to has not been associated to any of the supervisor namespaces. 
+    - **Fix**: Upload mentioned VM Image to content library and associate content library to your namespace. Refer to [VM Image](https://vm-operator.readthedocs.io/en/stable/concepts/images/) for details.
+- Reason `ContentSourceBindingNotFound`: this means that the content library which contains the specified VM Image has not been associated to the Supervisor namespace. 
+    - **Fix**: Associate content library to your namespace. Refer to [VM Image](https://vm-operator.readthedocs.io/en/stable/concepts/images/) for details.
+
+### GuestCustomization/VirtualMachineTools
+- Reason `GuestCustomizationFailed`/`VirtualMachineToolsNotRunning`:
+This could be happening when using transport `OvfEnv/ExtraConfig` to deploy a VM. These two transports use VMTools for network configuration, using incompatible virtual machine images could cause race with cloud-init user-data configuration. 
+    - **Fix**: use transport `CloudInit`. Refer to [Deploy with CloudInit](https://vm-operator.readthedocs.io/en/stable/tutorials/deploy-vm/cloudinit/) for details.
```

Comment:
I don't think this is an accurate solution, and we should also not limit the usage of VM Operator by proposing a closed solution in our docs.

Reply:
But for `OvfEnv/ExtraConfig` we are recommending `Any consumers still relying on this provider should switch to Cloud-Init.`
https://vm-operator.readthedocs.io/en/stable/tutorials/deploy-vm/#extraconfig

Reply:
@zyiyi11 
That's a recommendation going forward for `OvfEnv/ExtraConfig` to switch to `CloudInit` but the Conditions failing with '`GuestCustomizationFailed`/`VirtualMachineToolsNotRunning` doesn't always mean that the transport is the problem so we shouldn't be calling `Switching to CloudInit` as a **Fix** because there are other reason for the above conditions to happen